Content.Server/_DV/CosmicCult/MonumentSystem.cs

```diff
@@ -0,0 +1,369 @@
+    {
+        if (_cosmicRule.AssociatedGamerule(ent) is not {} cult)
+            return;
+        if (!TryComp<CosmicFinaleComponent>(ent, out var finaleComp))
+            return;
+        _appearance.SetData(ent, MonumentVisuals.Monument, cult.Comp.CurrentTier);
+
+        switch (cult.Comp.CurrentTier)
+        {
+            case 3:
+                _appearance.SetData(ent, MonumentVisuals.Tier3, true);
+                break;
+            case 2:
+                _appearance.SetData(ent, MonumentVisuals.Tier3, false);
+                break;
+        }
+
+        if (tierUp)
+        {
+            var transformComp = EnsureComp<MonumentTransformingComponent>(ent);
+            transformComp.EndTime = _timing.CurTime + ent.Comp.TransformTime;
+            _appearance.SetData(ent, MonumentVisuals.Transforming, true);
+        }
+
+        if (finaleComp.CurrentState != FinaleState.Unavailable)
+            _appearance.SetData(ent, MonumentVisuals.FinaleReached, true);
+    }
+
+    //note - these ar the thresholds for moving to the next tier
+    //so t1 -> 2 needs 20% of the crew
+    //t2 -> 3 needs 40%
+    //and t3 -> finale needs an extra 20 entropy
+    public void UpdateMonumentReqsForTier(Entity<MonumentComponent> monument, int tier)
+    {
+        if (_cosmicRule.AssociatedGamerule(monument) is not {} cult)
+            return;
+
+        var tier3NumCrew = Math.Round((double)cult.Comp.TotalCrew / 100 * _config.GetCVar(DCCVars.CosmicCultTargetConversionPercent)); // 40% of current pop
+
+        switch (tier)
+        {
+            case 1:
+                monument.Comp.ProgressOffset = 0;
+                monument.Comp.TargetProgress = (int)(tier3NumCrew / 2 * _config.GetCVar(DCCVars.CosmicCultistEntropyValue));
+                break;
+            case 2:
+                monument.Comp.ProgressOffset = (int)(tier3NumCrew / 2 * _config.GetCVar(DCCVars.CosmicCultistEntropyValue)); //reset the progress offset
+                monument.Comp.TargetProgress = (int)(tier3NumCrew * _config.GetCVar(DCCVars.CosmicCultistEntropyValue));
+                break;
+            case 3:
+                monument.Comp.ProgressOffset = (int)(tier3NumCrew * _config.GetCVar(DCCVars.CosmicCultistEntropyValue));
+                monument.Comp.TargetProgress = (int)(tier3NumCrew * _config.GetCVar(DCCVars.CosmicCultistEntropyValue)); //removed offset; replaced with timer
+                break;
+        }
+    }
+
+    public void SetCanTierUp(Entity<MonumentComponent> ent, bool canTierUp)
+    {
+        ent.Comp.CanTierUp = canTierUp;
+    }
+
+    public void SetTargetProgess(Entity<MonumentComponent> ent, int targetProgress)
+    {
+        ent.Comp.TargetProgress = targetProgress;
+    }
+
+    public void Disable(Entity<MonumentComponent> ent)
+    {
+        ent.Comp.Enabled = false;
+    }
+
+    public void Enable(Entity<MonumentComponent> ent)
+    {
+        ent.Comp.Enabled = true;
+    }
+
```

Comment:
All four of these functions could be inlined or static

Reply:
methods are for complying with access

Reply:
no but static methods are a bit smelly in systems because we have dep injection and the caller shouldn't care about if the system method has any dependencies or not